### ISSUE TITLE
I wanted to try making infinite scrolling with server components, I almost got there I think, but got stuck when I realised that `router.push` no loger accepts `{ scroll: false }`, meaning I can't encode things like "how far has data been loaded" as a search param, without reseting scroll on every change.

### ISSUE THREAD

**Issue by alvarlagerlof** [vercel/next.js#43552]
I wanted to try making infinite scrolling with server components, I almost got there I think, but got stuck when I realised that `router.push` no loger accepts `{ scroll: false }`, meaning I can't encode things like "how far has data been loaded" as a search param, without reseting scroll on every change.

Is this intentional? If so, have their been considerations about alternatives?

Example: https://next-app-server-infinite-scroll.vercel.app/
Repo: https://github.com/alvarlagerlof/next-app-server-infinite-scroll

_Originally posted by @alvarlagerlof in https://github.com/vercel/next.js/discussions/41745#discussioncomment-4103276_

**Comment by alvarlagerlof** [comment]
Hmm, maybe this "Reference in new issue" feature shortcuted the templates. Let me know if I should redo this.

**Comment by balazsorban44** [comment]
Yeah, the "Reference in new issue" feature has quite a bad UX. Since you added a reproduction it should be fine, but in the future, it might be better to create a new issue via [the template](https://github.com/vercel/next.js/issues/new?assignees=&labels=template%3A+bug&template=1.bug_report.yml) and manually reference it in the discussion. :pray:

**Comment by alvarlagerlof** [comment]
> Yeah, the "Reference in new issue" feature has quite a bad UX. Since you added a reproduction it should be fine, but in the future, it might be better to create a new issue via [the template](https://github.com/vercel/next.js/issues/new?assignees=&labels=template%3A+bug&template=1.bug_report.yml) and manually reference it in the discussion. :pray:

Yeah I noticed. I will keep that in mind for the future.

**Comment by steve-marmalade** [comment]
Commenting in the hope it helps prevent this issue from getting lost in the shuffle. Definitely feels like an important use-case.

**Comment by Gyurmatag** [comment]
What's up with this issue?

**Comment by Gyurmatag** [comment]
Hi @balazsorban44 !
Can you please tell us when this bug will be resolved? It is currently a blocking thing in our application.
Köszi szépen! :)

**Comment by steve-marmalade** [comment]
I periodically check out https://next-app-server-infinite-scroll.vercel.app to see whether updates to NextJS are improving RSC support for infinite scroll.

It looks to me like 13.3.1 not only fixes a regression introduced in 13.3.0 where new search params were causing client-side state to be lost (and so each new `?page=<n>` was scrolling up to the top), I'm also now seeing scroll position preserved when hitting the browser back button after clicking on one of the cards in the feed (which I believe is the crux of this Github Issue).

**Comment by steve-marmalade** [comment]
> Now it is broken again.

Yes, it looks like the issue is outlined in https://github.com/vercel/next.js/issues/49087

**Comment by timneutkens** [comment]
Duplicate of https://github.com/vercel/next.js/issues/49087
